Add possibility to have unembedded GeoJSON.

BibMartin opened the pull request:
Addresses #285.
Instead of having an `embed` option as @ocefpaf suggested, the user shall pass `data=open(filename)` if he wants to have the file embedded, and `data=filename` otherwise.

Note that the `style_function` cannot be applyed in the latter case, because the file won't be openned.

ocefpaf commented:
> Note that the `style_function` cannot be applied in the latter case, because the file won't be opened.

How difficult would it be to read the style from the file and still not embed?  We do not need to do this in this PR, but I want to keep that idea around.

ocefpaf commented on folium/features.py:
```diff
@@ -227,18 +229,20 @@ def __init__(self, data, style_function=None):
         super(GeoJson, self).__init__()
         self._name = 'GeoJson'
         if 'read' in dir(data):
```
I am not opposed to this pattern: `read in dir(x)`, but I think `hasattr(object, name)` is more common?  Any special reason for doing that?

ocefpaf commented on folium/features.py:
```diff
@@ -227,18 +229,20 @@ def __init__(self, data, style_function=None):
         if 'read' in dir(data):
+            self.embed = True
             self.data = json.load(data)
         elif type(data) is dict:
```
Same as above.  I see `isinstance(object, dict)` more that `type(object) is dict`.  Again, not opposed, just curious.

BibMartin commented:
@ocefpaf 
> I am not opposed to this pattern: read in dir(x), but I think hasattr(object, name) is more common? Any special reason for doing that?

I guess this has few importance ; I'm just a goldfish who forgot the clean syntax. Corrected.

> Same as above. I see isinstance(object, dict) more that type(object) is dict. Again, not opposed, just curious.

This one is more important ; for instance, someone may be willing to provide an `OrderedDict` instead of a `dict` ; yet `type(OrderedDict()) is dict` is False while `isinstance(OrderedDict(),dict)` is `True`. So, your (official) syntax is better.

I've corrected both lines.

ocefpaf commented:
> I guess this has few importance

I just asked for shear curiosity.  Sometimes I learn new things by asking :wink: 

> So, your (official) syntax is better.

I suspected so, but haven't tested it.  Good to know!

Thanks!  Merging this...
